### PR TITLE
Make fewer assumptions about log handling

### PR DESCRIPTION
**Pull request by jglick**
Needed for https://github.com/jenkinsci/workflow-job-plugin/pull/27, but generally should be an improvement.

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/test/BuildWatcher.java [hunk 1]
```diff
@@ -90,7 +89,7 @@
                 return;
             }
             RunningBuild build = new RunningBuild(r);
-            RunningBuild orig = builds.put(r.getLogFile(), build);
+            RunningBuild orig = builds.put(r.getRootDir(), build);
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/BuildWatcher.java]: We just needed a unique identifier of a build × `$JENKINS_HOME`; were not actually using the log file per se.

File: src/main/java/org/jvnet/hudson/test/BuildWatcher.java [hunk 2]
```diff
@@ -112,18 +111,18 @@
...
         RunningBuild(Run<?,?> r) {
-            log = r.getLogText();
+            this.r = r;
             sink = new LogLinePrefixOutputFilter(System.err, "[" + r + "] ");
         }
 
         synchronized void copy() {
             try {
-                pos = log.writeLogTo(pos, sink);
+                pos = r.getLogText().writeLogTo(pos, sink);
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/BuildWatcher.java]: See [this precedent](https://github.com/jenkinsci/jenkins/blob/3f5fe7f6b9b8dab37e82b360574102c6388729e8/core/src/main/java/hudson/cli/ConsoleCommand.java#L75-L79). `getLogText` in general needs to be called repeatedly as more log output arrives.

File: src/main/java/org/jvnet/hudson/test/BuildWatcher.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/java/org/jvnet/hudson/test/BuildWatcher.java]: Won't help with ExternalTaskLogging story

**Comment by oleg-nenashev** [comment]
🐝

File: src/main/java/org/jvnet/hudson/test/BuildWatcher.java (hunk 1 above)
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/BuildWatcher.java]: If you are referring to logging from non-`Run`s, that is true, but anyway `BuildWatcher` does not support them currently either.

**Pull request merged**